Packs/Illumio/Integrations/IllumioCore/IllumioCore_test.py

```diff
@@ -338,3 +378,207 @@ def test_virtual_service_create_command_when_invalid_arguments_provided(err_msg,
+
+    Given:
+        - object_provision_command function and mock_client to call the function
+    When:
+        - Invalid arguments (Empty values) provided in the arguments
+    Then:
+        - Should raise ValueError with proper error message
+    """
+    from IllumioCore import object_provision_command
+
+    args = {"security_policy_objects": ""}
+    err_msg = (
+        "security_policy_objects is a required parameter. Please provide correct value."
+    )
+
+    with pytest.raises(ValueError) as err:
+        object_provision_command(mock_client, args)
+
+    assert str(err.value) == err_msg
+
+
+def test_object_provision_command_when_valid_arguments_provided(
+    mock_client, object_provision_success, monkeypatch
+):
+    """
+    Test case scenario for execution of object_provision_command when valid arguments are provided.
+
+    Given:
+        - object_provision_command function and mock_client to call the function
+    When:
+        - Valid href is provided in the command argument
+    Then:
+        - Should return proper raw_response
+    """
+    monkeypatch.setattr(
+        illumio.pce.PolicyComputeEngine,
+        "provision_policy_changes",
+        lambda *a, **k: PolicyVersion.from_json(object_provision_success),
+    )
+    mock_args = {"security_policy_objects": "/orgs/1/sec_policy/draft/rule_sets/1605"}
+    response = object_provision_command(mock_client, mock_args)
+
+    assert response.raw_response == object_provision_success
+
+
+def test_object_provision_command_when_valid_arguments_provided_human_readable(
+    mock_client, object_provision_success, object_provision_success_hr
+):
+    """
+    Test case scenario for execution of object_provision_command when valid arguments are provided.
+
+    Given:
+        - object_provision_command function and mock_client to call the function
+    When:
+        - Valid href is provided in the command argument
+    Then:
+        - Returns a valid human-readable
+    """
+    resp = prepare_object_provision_output(object_provision_success)
+    assert resp == object_provision_success_hr
+
+
+def test_object_provision_command_when_406_error_code_returned(
+    mock_client, requests_mock
+):
+    """
+    Test case scenario for object_provision_command when object to be provision is already provisioned.
+
+    Given:
+        - object_provision_command function and mock_client to call the function
+    When:
+        - Object to be provision is already provisioned
+    Then:
+        - Should raise exception with proper error message
+    """
+    from IllumioCore import object_provision_command
+
+    mock_args = {"security_policy_objects": "/orgs/1/sec_policy/draft/rule_sets/1605"}
+    expected_sdk_response = util_load_json(
+        os.path.join(TEST_DATA_DIRECTORY, "object_provision_failure.json")
+    )
+
+    expected_err_msg = "406 Client Error: None for url: https://127.0.0.1:8443/api/v2/orgs/1/sec_policy"
```

Comment:
The None here is due to not providing a `reason` field in the call to `requests_mock.post` -  is there value in testing the error message here?